applets/wncklet/window-list.c

```diff
@@ -252,15 +252,18 @@ preview_window_thumbnail (WnckWindow   *wnck_window,
                           int          *thumbnail_scale)
 {
 	GdkWindow *window;
+	Window win;
 	cairo_surface_t *thumbnail;
 	cairo_t *cr;
 	double ratio;
 	int width, height, scale;
 
-	window = gdk_x11_window_foreign_new_for_display (gdk_display_get_default (), wnck_window_get_xid (wnck_window));
-
-	if (window == NULL)
-		return NULL;
+	win = wnck_window_get_xid (wnck_window);
+	if ((window = gdk_x11_window_lookup_for_display (gdk_display_get_default (), win)) == NULL)
```

Comment:
This function is already called by `gdk_x11_window_foreign_new_for_display`: https://gitlab.gnome.org/GNOME/gtk/-/blob/3.24.24/gdk/x11/gdkwindow-x11.c#L1244-1245

I don't think this PR changes anything, it just forces the lookup function to be called twice.

Reply:
The thumbnail of a window has been updated the second time I hovered the mouse over the taskbar.